makeflow/src/dag_file.h

```diff
@@ -16,16 +16,32 @@ See the file COPYING for details.
  * unique node, if any, that produces the file.
  */
 
+typedef enum {
+	DAG_FILE_STATE_UNKNOWN = 0,
+	DAG_FILE_STATE_EXPECT = 1,
+	DAG_FILE_STATE_EXISTS = 2,
+	DAG_FILE_STATE_COMPLETE = 3,
+	DAG_FILE_STATE_DELETE = 4,
+	DAG_FILE_STATE_DOWN = 5,
+	DAG_FILE_STATE_UP = 6
+} dag_file_state_t;
+
+
 struct dag_file {
 	const char *filename;
 	struct list     *needed_by;              /* List of nodes that have this file as a source */
 	struct dag_node *created_by;             /* The node (if any) that created the file */
 	int    ref_count;                        /* How many nodes still to run need this file */
+	time_t creation_logged;                  /* Time that file creation is logged */
```

Comment:
This value is there so that when we log its creation we know the time stamp, which is used when recovering the log to determine is a file was modified between runs.